File: Algorithms/src/1975. Maximum Matrix Sum/Code/solution.cpp
class Solution
{
public:
    long long maxMatrixSum(vector<vector<int>> &matrix)
    {
        long long sum = 0;
        int negativeCount = 0;
        int minAbs = INT_MAX;

        for (auto &row : matrix)
        {
            for (int val : row)
            {
                sum += llabs(val); // add absolute value
                if (val < 0)
                    negativeCount++;
                minAbs = min(minAbs, abs(val));
            }
        }

        // If odd negatives, one smallest value must remain negative
        if (negativeCount % 2 == 1)
        {
            sum -= 2LL * minAbs;
        }

        return sum;
    }
};

File: Algorithms/src/1975. Maximum Matrix Sum/Code/solution.py
class Solution:
    def maxMatrixSum(self, matrix):
        total_sum = 0
        negative_count = 0
        min_abs = float('inf')

        for row in matrix:
            for val in row:
                total_sum += abs(val)      # add absolute value
                if val < 0:
                    negative_count += 1
                min_abs = min(min_abs, abs(val))

        if negative_count % 2 == 1:
            total_sum -= 2 * min_abs

        return total_sum
